rocketpy/tools.py

```diff
@@ -23,3 +26,978 @@ def __get__(self, instance, owner=None):
+    (1, 2, 3)
+    >>> v[0], v[1], v[2]
+    (1, 2, 3)
+
+    Vector instances can be added, subtracted, multiplied by a scalar, divided
+    by a scalar, negated, and cross and dot product can be computed:
+    >>> v + v
+    Vector(2, 4, 6)
+    >>> v - v
+    Vector(0, 0, 0)
+    >>> v * 2
+    Vector(2, 4, 6)
+    >>> v / 2
+    Vector(0.5, 1.0, 1.5)
+    >>> -v
+    Vector(-1, -2, -3)
+    >>> v @ v # Dot product
+    14
+
+    Cross products need to be wrapped in parentheses to ensure the ^ operator
+    precedence:
+    >>> (v ^ v)
+    Vector(0, 0, 0)
+
+    Vector instances can be called as functions if their elements are callable:
+    >>> v = Vector([lambda x: x**2, lambda x: x**3, lambda x: x**4])
+    >>> v(2)
+    Vector(4, 8, 16)
+
+    Vector instances magnitudes can be accessed as its absolute value:
+    >>> v = Vector([1, 2, 3])
+    >>> abs(v)
+    3.7416573867739413
+
+    Vector instances can be normalized:
+    >>> v.unit_vector
+    Vector(0.2672612419124244, 0.5345224838248488, 0.8017837257372732)
+
+    Vector instances can be compared for equality:
+    >>> v = Vector([1, 2, 3])
+    >>> u = Vector([1, 2, 3])
+    >>> v == u
+    True
+    >>> v != u
+    False
+
+    And last, but not least, it is also possible to check if two vectors are
+    parallel or orthogonal:
+    >>> v = Vector([1, 2, 3])
+    >>> u = Vector([2, 4, 6])
+    >>> v.is_parallel_to(u)
+    True
+    >>> v.is_orthogonal_to(u)
+    False
+    """
+
+    def __init__(self, components):
+        """Vector class constructor.
+
+        Parameters
+        ----------
+        components : array-like, iterable
+            An iterable with length equal to 3, corresponding to x, y and z
+            components.
+
+        Examples
+        --------
+        >>> v = Vector([1, 2, 3])
+        >>> v
+        Vector(1, 2, 3)
+        """
+        self.components = components
+
+    def __getitem__(self, i):
+        """Access vector components by indexing."""
+        return self.components[i]
+
+    def __call__(self, *args):
+        """Adds support for calling a vector as a function, if its elements are
+        callable.
+
+        Parameters
+        ----------
+        args : arguments
+            Arguments to be passed to the vector elements.
+
+        Returns
+        -------
+        Vector
+            Vector with the return of each element called with the given
+            arguments.
+
+        Examples
+        --------
+        >>> v = Vector([lambda x: x**2, lambda x: x**3, lambda x: x**4])
+        >>> v(2)
+        Vector(4, 8, 16)
+        """
+        try:
+            return self.element_wise(lambda f: f(*args))
+        except TypeError as exc:
+            msg = "One or more elements of this vector is not callable."
+            raise TypeError(msg) from exc
+
+    def __len__(self):
+        return 3
+
+    @cached_property
+    def x(self):
+        """First component of the vector."""
+        return self.components[0]
```

Comment:
If all relevant operations with Vectors involves this value (that will be eventually cached as an attribute) I would remove the small delay of the first caching and input `self.x` as a plain attribute on `__init__`.

Also, I would deconstruct the list so as not to call `__getitem__` thrice:

```py
self.x, self.y, self.z = components
```

Reply:
Great suggestion! I have implemented it in the new commits.